packager/hls/base/media_playlist.cc

```diff
@@ -423,13 +448,15 @@ void MediaPlaylist::AddSegment(const std::string& file_name,
                                          : std::next(iter)->timestamp;
       AddSegmentInfoEntry(file_name, iter->timestamp,
                           next_timestamp - iter->timestamp,
-                          iter->start_byte_offset, iter->size);
+                          iter->start_byte_offset, iter->size, start_timestamp);
     }
     key_frames_.clear();
-    return;
+  } else {
+    AddSegmentInfoEntry(file_name, start_time, duration, start_byte_offset, size,
+                        start_timestamp);
```

Comment:
Should we insert it for first segment and for the segments after ext-x-discontinuity only?

Reply:
We CAN insert it everywhere only once or more but if we did it then we MUST insert it after all ext-x-discontinuity. So, we CAN simply insert it for all segments, as far as I understand RFC 8216.

Reply:
Sure. I think it will be simpler if we just consider it for every ext-x-discontinuity. I am fine with your approach too but you'll need to take care of the i-frame playlist too. See the comment above.

Reply:
OK